helpers/UserHelper.php

```diff
@@ -153,13 +151,27 @@ public static function getUserName(User $user, $defaultToLabel = false)
     {
         $firstName = self::getUserStringProp($user, GenerisRdf::PROPERTY_USER_FIRSTNAME);
         $lastName = self::getUserStringProp($user, GenerisRdf::PROPERTY_USER_LASTNAME);
-        
+
         $userName = trim($firstName . ' ' . $lastName);
-        
+
         if (empty($userName) && $defaultToLabel) {
             $userName = self::getUserLabel($user);
         }
 
         return $userName;
     }
+
+    /**
+     * @param User|string $user
+     *
+     * @return string
+     */
+    public static function getUserUILanguageCode($user): string
+    {
+        if (is_string($user)) {
+            $user = self::getUser($user);
+        }
+
```

Comment:
Do you think we could have strict types as type hint? Mixed type generally opens a precedent to increase coupling and difficult to maintain / refactor the method after.

```suggestion
    public static function getUserUILanguageCode(string $userId): string
    {
        $user = self::getUser($userId);

```

Reply:
Do you think we should have this type of logic here instead? 

https://github.com/oat-sa/generis/blob/master/common/oatbox/user/UserLanguageService.php